qiskit/visualization/circuit/matplotlib.py

```diff
@@ -1337,39 +1359,58 @@ def _measure(self, node, node_data, outer_circuit, glob_data):
             zorder=PORDER_GATE,
         )
         self._ax.add_patch(arc)
+        qy_adj2 = 0.2 if self._measure_arrows else 0.3
         self._ax.plot(
             [qx, qx + 0.35 * WID],
-            [qy - 0.15 * HIG, qy + 0.20 * HIG],
+            [qy - qy_adj1 * HIG, qy + qy_adj2 * HIG],
             color=node_data[node].gt,
             linewidth=self._lwidth2,
             zorder=PORDER_GATE,
         )
-        # arrow
-        self._line(
-            node_data[node].q_xy[0],
-            [cx, cy + 0.35 * WID],
-            lc=self._style["cc"],
-            ls=self._style["cline"],
-        )
-        arrowhead = glob_data["patches_mod"].Polygon(
-            (
-                (cx - 0.20 * WID, cy + 0.35 * WID),
-                (cx + 0.20 * WID, cy + 0.35 * WID),
-                (cx, cy + 0.04),
-            ),
-            fc=self._style["cc"],
-            ec=None,
-        )
-        self._ax.add_artist(arrowhead)
-        # target
-        if self._cregbundle and register is not None:
+        # If measure_arrows, draw the down arrow to the clbit
+        if self._measure_arrows:
+            self._line(
+                node_data[node].q_xy[0],
+                [cx, cy + 0.35 * WID],
+                lc=self._style["cc"],
+                ls=self._style["cline"],
+            )
+            arrowhead = glob_data["patches_mod"].Polygon(
+                (
+                    (cx - 0.20 * WID, cy + 0.35 * WID),
+                    (cx + 0.20 * WID, cy + 0.35 * WID),
+                    (cx, cy + 0.04),
+                ),
+                fc=self._style["cc"],
+                ec=None,
+            )
+            self._ax.add_artist(arrowhead)
+            # target
+            if self._cregbundle and register is not None:
+                self._ax.text(
+                    cx + 0.25,
+                    cy + 0.1,
+                    str(reg_index),
+                    ha="left",
+                    va="bottom",
+                    fontsize=0.8 * self._style["fs"],
+                    color=self._style["tc"],
+                    clip_on=True,
+                    zorder=PORDER_TEXT,
+                )
+        else:
+            # If not measure_arrows, write the reg_bit into the measure box
+            if register is not None:
+                label = f"{register.name}_{reg_index}"
```

Comment:
Would it be better to use something like:
```py
                label = f"{register.name}[{reg_index}]"
```
As it represents the relationship better. This should apply to the text drawer counterpart as well.